tests/main/services-restart/task.yaml

```diff
@@ -0,0 +1,35 @@
+summary: Check that snap restarts affects only running services
+
+execute: |
+    echo "Installing the service snap"
+
+    # we are using systemd-notify indicate the service is active, this is
+    # currently not allowed by daemon-notify interface, so we may as well just
+    # install in devmode
+    "$TESTSTOOLS"/snaps-state install-local test-snapd-service-restart --devmode
+
+    echo "We can see all services running"
+    for id in 1 2 3 4; do
+        systemctl status snap.test-snapd-service-restart.service$id | MATCH "running"
+    done
+
+    echo "Stopping services 3 and 4"
+    snap stop test-snapd-service-restart.service3
+    snap stop test-snapd-service-restart.service4
```

Comment:
We should arrange service states to have:
- 1 enable service that is running
- 1 enable service that is stopped
- 1 disabled service that is running
- 1 disabled service that is stopped

Reply:
Isn't that the case already?

Reply:
Ah right, sorry missed that. Maybe a comment right before `echo "Stopping .."` in the test would be useful, with something like:
```
# arrange services such that their states are:
# service1 - disabled and running
# service2 - enabled and running
# service3 - disabled and stopped
# service4 - enabled and stopped
```

Reply:
Done!